src/vmm/tests/integration_tests.rs

```diff
@@ -327,6 +327,7 @@ fn test_snapshot_load_sanity_checks() {
     );
 
     // Create MAX_SUPPORTED_VCPUS vCPUs starting from 1 vCPU.
+    #[allow(clippy::cast_sign_loss)] // there is no impl TryFrom<f64> for usize
     for _ in 0..f64::from(MAX_SUPPORTED_VCPUS).log2() as usize {
```

Comment:
You can use [`ilog2`](https://doc.rust-lang.org/std/primitive.usize.html#method.ilog2) here.
```suggestion
        for _ in 0..MAX_SUPPORTED_VCPUS.ilog2() {
```

Reply:
oh, nice!